File: desktop/src/main/java/dev/ultreon/quantum/desktop/StatusCode.java
package dev.ultreon.quantum.desktop;

import dev.ultreon.quantum.server.PlatformOS;

public class StatusCode {
    public static int forAbort() {
        if (PlatformOS.isWindows) {
            return 3;
        } else if (PlatformOS.isMac) {
            return 6;
        } else if (PlatformOS.isLinux) {
            return 6;
        }
        return -1;
    }

    public static int forException() {
        return PlatformOS.isWindows ? -1 : 1;
    }
}

File: desktop/src/main/java/dev/ultreon/quantum/desktop/StatusCode.kt
package dev.ultreon.quantum.desktop

import dev.ultreon.quantum.OS

object StatusCode {
  @JvmStatic
  fun forAbort(): Int {
    if (OS.isWindows()) {
      return 3
    } else if (OS.isMac()) {
      return 6
    } else if (OS.isLinux()) {
      return 6
    }
    return -1
  }

  @JvmStatic
  fun forException(): Int {
    return if (OS.isWindows()) -1 else 1
  }
}
